tikv-client/src/main/java/com/pingcap/tikv/expression/visitor/ProtoConverter.java

```diff
@@ -308,16 +326,36 @@ protected Expr visit(IsNull node, Object context) {
     ScalarFuncSig protoSig = ScalarFuncSig.valueOf(typeSignature + "IsNull");
     Expr.Builder builder = scalarToPartialProto(node, context);
     builder.setSig(protoSig);
-    builder.setFieldType(FieldType.newBuilder().setTp(getType(node).getTypeCode()).build());
+    builder.setFieldType(toPBFieldType(getType(node)));
     return builder.build();
   }
 
   @Override
   protected Expr visit(Not node, Object context) {
-    ScalarFuncSig protoSig = ScalarFuncSig.UnaryNot;
+    ScalarFuncSig protoSig = null;
+    DataType dataType = getType(node);
+    switch (dataType.getType()) {
```

Comment:
how about we add a link to tipb?

Reply:
The difference is caused by upgrading tipb.